src/main/java/org/springframework/data/elasticsearch/client/reactive/DefaultReactiveElasticsearchClient.java

```diff
@@ -401,6 +404,17 @@ public Indices indices() {
 				.flatMap(Flux::fromIterable);
 	}
 
+	/*
+	 * (non-Javadoc)
+	 * @see org.springframework.data.elasticsearch.client.reactive.ReactiveElasticsearchClient#aggregate(org.springframework.http.HttpHeaders, org.elasticsearch.action.search.SearchRequest)
+	 */
+	@Override
+	public Flux<Aggregation> aggregate(HttpHeaders headers, SearchRequest searchRequest) {
+		return sendRequest(searchRequest, requestCreator.search(), SearchResponse.class, headers) //
```

Comment:
```suggestion
		
		Assert.notNull(headers, "headers must not be null");
		Assert.notNull(searchRequest, "searchRequest must not be null");
		
		searchRequest.source().size(0);
		
		return sendRequest(searchRequest, requestCreator.search(), SearchResponse.class, headers) //
```

we should check non-nullable parameters and as we don't need the source in the response, we should set it to zero